test/e2e/specs/site-editor/style-variations.spec.js

```diff
@@ -0,0 +1,266 @@
+		).toHaveCSS( 'background', /rgb\(74, 7, 74\)/ );
+
+		await expect(
+			page.locator( 'role=button[name="Color: Background"i]' )
+		).toHaveCSS( 'background', /rgb\(202, 105, 211\)/ );
+
+		await expect(
+			page.locator( 'role=button[name="Color: Awesome pink"i]' )
+		).toHaveCSS( 'background', /rgba\(204, 0, 255, 0\.77\)/ );
+	} );
+
+	test( 'should reflect style variations in the styles applied to the editor', async ( {
+		admin,
+		page,
+		siteEditorStyleVariations,
+	} ) => {
+		await admin.visitSiteEditor( {
+			postId: 'gutenberg-test-themes/style-variations//index',
+			postType: 'wp_template',
+		} );
+		await siteEditorStyleVariations.applyYellowVariation();
+		const frame = page.frameLocator(
+			'css=.edit-site-visual-editor iframe'
+		);
+		const paragraph = frame.locator(
+			'xpath=//p[text()="My awesome paragraph"]'
+		);
+		const paragraphColor = await paragraph.evaluate( ( el ) => {
+			return window.getComputedStyle( el ).color;
+		} );
+		expect( paragraphColor ).toBe( 'rgb(25, 25, 17)' );
+		const body = frame.locator( 'css=body' );
+		const backgroundColor = await body.evaluate( ( el ) => {
+			return window.getComputedStyle( el ).backgroundColor;
+		} );
+		expect( backgroundColor ).toBe( 'rgb(255, 239, 11)' );
+	} );
+} );
+
+class SiteEditorStyleVariations {
+	constructor( { admin, editor, page, pageUtils, requestUtils } ) {
+		this.admin = admin;
+		this.editor = editor;
+		this.page = page;
+		this.pageUtils = pageUtils;
+		this.requestUtils = requestUtils;
+	}
+
+	async openOtherStyles() {
+		await this.page
+			.locator( 'role=button[name="Browse styles"i]' )
+			.waitFor( { state: 'attached' } ); // Wait for the element to appear after the navigation animation.
+
+		await this.page.click( 'role=button[name="Browse styles"i]' );
+	}
+
+	async applyVariation( label ) {
+		await this.page.click( 'role=button[name="Styles"i]' );
+		await this.openOtherStyles();
+		await this.page.click( `role=button[name="${ label }"i]` );
+	}
+
+	async applyPinkVariation() {
+		await this.applyVariation( 'pink' );
+	}
+
+	async applyYellowVariation() {
+		await this.applyVariation( 'yellow' );
+	}
+
+	async openColorsPanel() {
+		await this.openGlobalStylesPanel( 'Colors styles' );
+	}
+
+	async openTypographyPanel() {
+		await this.openGlobalStylesPanel( 'Typography styles' );
+	}
+
+	async openTextPanel() {
+		await this.openGlobalStylesPanel( 'Typography Text styles' );
+	}
+
+	async getFontSizeHint() {
+		const element = this.page.locator(
+			'css=.components-font-size-picker__header__hint'
+		);
+		return await element.evaluate( ( el ) => el.textContent );
+	}
+
+	async getCustomFontSizeValue() {
+		const element = this.page.locator(
+			'css=.components-font-size-picker input[aria-label="Custom"]'
+		);
+		return await element.evaluate( ( el ) => el.value );
+	}
+
+	async openPreviousGlobalStylesPanel() {
+		await this.page.click(
+			'role=button[name="Navigate to the previous view"i]'
+		);
+	}
```

Comment:
Since that role selectors are already pretty readable, I'd suggest inlining this directly to avoid the readers having to jump between the utils and their implementations that often.

Same for other one/two-line utils in this file.